lib/actions/local/common.js

```diff
@@ -0,0 +1,99 @@
+import * as Promise from 'bluebird';
```

Comment:
This file will be transpiled like any typescript files, which allows us to use modern javascript functionality like async/await without having to worry about node 8 supporting it or not - I haven't made use of any of it outside of modules because I wanted this to be a minimalistic conversion for maximum safety.  
This file will also be type checked as much as possible and linted correctly (with actual lint errors unlike for coffeescript where it's only warnings)